source/configuration_manual/mail_location/imapc/index.rst

```diff
@@ -100,24 +111,35 @@ Connection Settings
 
    **Optimizations**
 
-   ``fetch-bodystructure``
+   ``no-fetch-bodystructure``
+
+     Disable fetching of ``BODY`` and ``BODYSTRUCTURE`` which is available by
+     default.
+
+     .. versionchanged:: v2.4.0;v3.0.0 Earlier versions had a
+                         "fetch-bodystructure" feature, which is now enabled by
+                         default.
+
+   ``no-fetch-headers``
 
-     Allow fetching ``BODY`` and ``BODYSTRUCTURE``
-     ``FETCH BODY.PEEK[HEADER.FIELDS (..)]``.
+     Disable fetching of specific message headers from remote server.
```

Comment:
Disable fetching of specific message headers from remote server using the IMAP ``FETCH BODY.PEEK[HEADER.FIELDS (...)]`` command. Instead, the whole header is fetched and the wanted headers are parsed from it.